### PR TITLE
feature: integrate publication feature

### PR DESCRIPTION
**Pull request by codeWhizperer**

### CODE REVIEW

File: src/publication/publication.cairo [hunk 1]
```diff
@@ -288,38 +381,71 @@ pub mod PublicationComponent {
...
+            let is_channel_member = self
+                ._is_channel_member(get_caller_address(), collect_params.channel_id);
+            assert(community_ban_status == false, BANNED_MEMBER);
+            assert(channel_ban_status == false, BANNED_FROM_CHANNEL);
+            assert(is_community_member == true, NOT_COMMUNITY_MEMBER);
+            assert(is_channel_member == true, NOT_CHANNEL_MEMBER);
             let collect_nft_address = self
                 ._get_or_deploy_collect_nft(
-                    karst_hub, profile_address, pub_id, collect_nft_impl_class_hash, salt
+                    collect_params.karst_hub,
```
Comment (Darlington02) [inline src/publication/publication.cairo:437]: 80% of the parameters passed into `collect_params` do not make sense to let users pass in. karst_hub, impl_hash, salt. even profile_address should be gotten from get_caller_address. please remove collectparams type, and store what needs to be stored in storage. for salt you can use something random like the profile_address collecting or timestamp.

File: tests/test_publication.cairo [hunk 2]
```diff
@@ -51,96 +58,48 @@ fn __setup__() -> (
...
-    let user_one_profile_address = dispatcher
-        .create_profile(
-            nft_contract_address,
-            (*registry_class_hash.class_hash).into(),
-            (*account_class_hash.class_hash).into(),
-            2478
-        );
-    let content_URI: ByteArray = "ipfs://helloworld";
+    // spy
     let mut spy = spy_events();
```
Comment (Darlington02) [inline tests/test_publication.cairo:93]: my 2c is this does not belong in setup. complicates it a little bit, and is just one line of code so can easily go into the test fns themselves

File: tests/test_publication.cairo [hunk 3]
```diff
@@ -859,158 +863,396 @@ fn test_repost_event_emission() {
...
+        .get_publication(user_one_profile_address, pub_assigned_id)
         .collect_nft;
     let collect_dispatcher = ICollectNFTDispatcher { contract_address: collect_nft2 };
-    let user3_token_id = collect_dispatcher.get_user_token_id(user_three_profile_address);
+    let user3_token_id = collect_dispatcher.get_user_token_id(USER_THREE.try_into().unwrap());
     assert(collect_nft1 == collect_nft2, 'invalid_ address');
     assert(token_id2 == user3_token_id, 'invalid token_id');
     stop_cheat_caller_address(publication_contract_address);
 }
+
```
Comment (Darlington02) [inline tests/test_publication.cairo:1258]: Always good to include test for negative scenarios - e.g error thrown when user tries to post in community/channel he doesn't belong to
